src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionMemberAccessor.cs

```diff
@@ -8,6 +8,7 @@
 
 namespace System.Text.Json.Serialization.Metadata
 {
+    [RequiresDynamicCode("Runtime code generation is not available for Aot form factor.")]
```

Comment:
`ReflectionMemberAccessor` should work fine with AOT (as long as the code is not stripped).

Where is the `ReflectionMemberAccessor` using AOT unfriendly code?

Reply:
This was due to the side effect of adding the warning to the base type to get rid of the `IL3051` [warning](https://github.com/dotnet/runtime/pull/67582#discussion_r843402565). Will add a suppression to `ReflectionMemberAccessor`.

Reply:
The other places in this PR are on plan to only annotate the constructor. Would it make sense to follow the same plan here? Something like:
- MemberAccessor - no annotations on the whole type
- ReflectionMemberAccessor - no annotations on the whole type
- ReflectionEmitMemberAccessor - constructor annotated, the rest is suppressed with a comment that constructor is annotated